docs/src/tutorial.rst

```diff
@@ -0,0 +1,496 @@
+
+This creates *my_widget_service.ts* in the *bin* directory.
+We don't need most of this code,
+so for now change it to the following:
+
+.. code-block:: ts
+
+    #!/usr/bin/env node
+    import cdk = require('@aws-cdk/cdk');
+
+    class MyWidgetServiceStack extends cdk.Stack {
+      constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+        super(parent, name, props);
+
+
+      }
+    }
+
+    // Create a new CDK app
+    const app = new cdk.App();
+
+    // Add your stack to it
+    new MyWidgetServiceStack(app, 'MyWidgetServiceStack');
+
+    app.run();
+
+Save it and make sure it builds and creates an empty stack.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+You should see a stack like the following,
+where CDK-VERSION is the version of the CDK.
+
+.. code-block:: sh
+
+    Resources:
+      CDKMetadata:
+        Type: 'AWS::CDK::Metadata'
+        Properties:
+          Modules: >-
+            @aws-cdk/cdk=CDK-VERSION,@aws-cdk/cx-api=CDK-VERSION,my_widget_service=0.1.0
+
+.. _create_lambda_functions:
+
+Step 2: Create a |LAM| Function to List all Widgets
+===================================================
+
+The next step is to create a |LAM| function to list all of the widgets in our
+|S3| bucket.
+
+Create the directory *resource* at the same level as the *bin* directory.
+
+.. code-block:: sh
+
+    mkdir resources
+
+Create the following Javascript file, *widgets.js*,
+in the *resources* directory.
+
+.. code-block:: js
+
+    const AWS = require('aws-sdk');
+    const S3 = new AWS.S3();
+
+    const bucketName = process.env.BUCKET;
+
+    // For now we only handle GET /:
+    exports.main = function(event, context, callback) {
+      if (event.httpMethod == "GET") {
+        if (event.path.length == 1) {
+          S3.listObjectsV2({ Bucket: bucketName })
+            .promise()
+            .then(function(data) {
+              var body = { widgets: data.Contents.map(function(e) { return e.Key }) };
+              var response = {
+                statusCode: 200,
+                headers: {},
+                body: JSON.stringify(body)
+              };
+              callback(null, response);
+            })
+            .catch(rejectedPromise);
+        }
+      }
+
+      function rejectedPromise(error) {
+        var body = error.stack || JSON.stringify(error, null, 2);
+        var response = makeError(body);
+        callback(null, response);
+      }
+    }
+
+Save it and make sure it builds and creates an empty stack.
+Note that since we haven't wired the function to our app,
+the Lambda file does not appear in the output.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _create_widgets_service:
+
+Step 3: Create Widgets Service
+==============================
+
+Add the |APIGATEWAY|, |LAM|, and |S3| packages to our app.
+
+.. code-block:: sh
+
+    npm install @aws-cdk/aws-apigateway
+    npm install @aws-cdk/aws-lambda
+    npm install @aws-cdk/aws-s3
```

Comment:
This is probably all possible on one line

Reply:
yep